modules/api/src/main/ForumAccess.scala

```diff
@@ -9,20 +9,35 @@ final class ForumAccess(teamApi: lila.team.TeamApi, teamCached: lila.team.Cached
     ec: scala.concurrent.ExecutionContext
 ) {
 
-  def isGrantedRead(categSlug: String)(implicit ctx: UserContext): Fu[Boolean] =
+  sealed trait Operation
+  case object Read  extends Operation
+  case object Write extends Operation
+
+  def isGranted(categSlug: String, op: Operation)(implicit ctx: UserContext): Fu[Boolean] =
     Categ.slugToTeamId(categSlug).fold(fuTrue) { teamId =>
       ctx.me ?? { me =>
-        fuccess(Granter(Permission.ModerateForum)(me)) >>| (teamApi
-          .belongsTo(teamId, me.id) >>& teamCached.forumAccess.get(teamId).flatMap {
-          case Team.Access.NONE    => fuFalse
-          case Team.Access.MEMBERS => fuTrue
+        fuccess(Granter(Permission.ModerateForum)(me)) >>| teamCached.forumAccess.get(teamId).flatMap {
+          case Team.Access.NONE => fuFalse
+          case Team.Access.EVERYONE =>
+            op match {
```

Comment:
this compiles? :thinking:

Reply:
strange, github shows me `on match {` but when I fetch the branch locally, my editor shows me `op match {`